outlook/src/taskpane/components/Log/Logger.tsx

```diff
@@ -27,36 +27,116 @@ class Logger extends React.Component<LoggerProps, LoggerState> {
         };
     }
 
+    private sendRequest =(requestJson: any) => {
+        const logRequest = sendHttpRequest(HttpVerb.POST, api.baseURL + api.logSingleMail, ContentType.Json,
+            this.context.getConnectionToken(), requestJson, true);
+        logRequest.promise.then(response => {
+            const parsed = JSON.parse(response);
+            if (parsed['error']) {
+                this.setState({logged: 0});
+                this.context.showHttpErrorMessage();
+                return;
+            } else {
+                this.setState({logged: 2});
+            }
+        }).catch(error => {
+            this.context.showHttpErrorMessage(error);
+            this.setState({logged: 0});
+        });
+    }
+
     private logRequest = (event) => {
         event.stopPropagation();
 
         this.setState({logged: 1});
         Office.context.mailbox.item.body.getAsync(Office.CoercionType.Html, (result) => {
             const msgHeader = '<div>From: ' + Office.context.mailbox.item.sender.emailAddress + '</div><br/>';
-            const msgFooter = '<br/><div class="text-muted font-italic">Logged from <a href="https://www.odoo.com/documentation/user/crm/optimize/mail_client_extension.html" target="_blank">Outlook Inbox</a></div>';
+            const msgFooter = '<br/><div class="text-muted font-italic">Logged from <a href="https://www.odoo.com/documentation/14.0/applications/sales/crm/optimize/outlook_extension.html" target="_blank">Outlook Inbox</a></div>';
             const body = result.value.split('<div id="x_appendonsend"></div>')[0]; // Remove the history and only log the most recent message.
             const message = msgHeader + body + msgFooter;
 
             const requestJson = {
                 res_id: this.props.resId,
                 model: this.props.model,
-                message: message
+                message: message,
+                attachments: [],
             }
-            const logRequest = sendHttpRequest(HttpVerb.POST, api.baseURL + api.logSingleMail, ContentType.Json,
-                this.context.getConnectionToken(), requestJson, true);
-            logRequest.promise.then(response => {
-                const parsed = JSON.parse(response);
-                if (parsed['error']) {
-                    this.setState({logged: 0});
-                    this.context.showHttpErrorMessage();
-                    return;
-                } else {
-                    this.setState({logged: 2});
+
+            const officeAttachmentDetails = Office.context.mailbox.item.attachments;
+            const promises: any[] = [];
+
+            if (officeAttachmentDetails.length > 0) {
+                for (let i = 0; i < officeAttachmentDetails.length; i++) {
+                    const promise = new Promise<any> ((resolve => {
```

Comment:
I think in es6 you can do something like:

```js
// grouping promises that allow fetching the actual content of the attachment then fetching them all
const officeAttachments = Office.context.mailbox.item.attachments;
const officeAttachmentsPromises = officeAttachments.map(officeAttachment => Office.context.mailbox.item.getAttachmentContentAsync(officeAttachment));
const attachmentContents = await Promise.all(officeAttachmentsPromises);

// format our attachments to be able to send them to Odoo
const formattedAttachments = attachmentContents.map(attachmentContent => {
    const formattedAttachment = {};
    // review comment: do all your necessary formatting, converting the "Outlook object" to your desired "Request ready object"
    return formattedAttachment;
})
```

Not sure exactly what that means for your "inline attachments", but you will figure it out :dagger: 

Would be a bit more readable and "pretty".

Also, when writing conditions, please format it this way:
```js
if (condition) {

}
```

Instead of:
```js
if (condition)
{

}
```

(I think I already mentioned this in another PR, installing and configuring a linter might help to warn you about such formatting issues wile you're coding).

Reply:
As discussed, the issue with the attachment content fetching is that it uses a callback and not a promise, for this reason we cannot use await/async for attachment contents fetching.

For inline attachments, we are obliged to use a counter in order to be able to map each inline attachment with the correspondent <img> tag.

Despite all this, I think that it's possible to make the code prettier by better structuring it and by for example using Arrays.forEach instead of loops wherever it's possible.

> (I think I already mentioned this in another PR, installing and configuring a linter might help to warn you about such formatting issues wile you're coding).

Linting issue finally solved, can you have a look again?